src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -32,7 +33,26 @@ public override string ToString()
                 return "\"{0}\"".FormatInvariant(stringValue);
             }
 
-            return this.ExpectedValue.ToString();
+            try
+            {
+                return this.ExpectedValue.ToString();
+            }
+            catch (Exception)
+            {
+                FakeManager manager;
+                try
+                {
+                    manager = Fake.GetFakeManager(this.ExpectedValue);
+                }
+                catch (ArgumentException)
+                {
+                    manager = null;
+                }

```

Comment:
I'd imagined a `TryGetFakeManager` when I made the comment. I've added one in a commit that I've [pushed to my own branch](https://github.com/blairconrad/FakeItEasy/commit/4c9c637ee35cbd2d46b1f75354f14446ad26dfae). If you like it, feel free to squash it in.
If not, I'm very happy with what you have here. I particularly like the tests. Thanks!